components/markdown_image.test.jsx

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {mountWithIntl} from 'tests/helpers/intl-test-helper.jsx';
+import configureStore from 'redux-mock-store';
+import {Provider} from 'react-redux';
+
+import MarkdownImage from 'components/markdown_image.jsx';
+import SizeAwareImage from 'components/size_aware_image';
+
+describe('components/MarkdownImage', () => {
+    const mockStore = configureStore();
+    const initialState = {
+        entities: {
+            general: {
+                config: {
+                    DefaultClientLocale: 'en',
+                },
+            },
+        },
+    };
+    const store = mockStore(initialState);
+    it('should match snapsnot', () => {
+        const wrapper = mountWithIntl(
+            <Provider store={store}>
+                <MarkdownImage imagesMetadata={{format: 'jpg', frame_count: 0, width: 100, height: 100}} src='path/image'/>
```

Comment:
Ideally i wanted to make this a shallow snapshot and find the component prop to have `dimensions` but i didn't do it with the component being a function. @hmhealey Any thoughts or suggestions on how i can do this.

Reply:
I'm not sure how to do it when the Provider is required, but I feel like it involves something like shallowly rendering the component and then using `.dive()` to render the contents of the `ExternalImage` ([link](https://airbnb.io/enzyme/docs/api/ShallowWrapper/dive.html)). I haven't tried that out myself, but that seems to be what we're looking for here

Reply:
Hi @sudheerDev , @hmhealey 

I've subscribed to this issue as I'm working on https://github.com/mattermost/mattermost-server/issues/9805 and will need to rebase after this one is merged 😄 

In my test file, I've used `dive`, here is a snippet from one of my tests 

```
test('should render a link if the source is unsafe', () => {
        const props = {...baseProps, src: '/images/logo.png'};
        const wrapper = shallow(
            <MarkdownImage {...props}/>
        );
        const childrenWrapper = shallow(wrapper.dive({context: {store}}).props().children());

        expect(childrenWrapper.find('a.markdown__link')).toHaveLength(1);
        expect(childrenWrapper).toMatchSnapshot();
    });
```

Reply:
@abdusabri Thanks for the suggestion. Updated the tests

Reply:
@abdusabri FYI, I realized that you don't actually need the dive there. You can just call `wrapper.props().children('link')` to have the same effect

Reply:
Thanks for the clarification @hmhealey , it is indeed not needed 🙂. My code snippet was from an earlier iteration where i had the markdown_image connected to the store - no longer the case though -, and the dive was needed then. My bad that I didn't write back to clarify when i changed my code 🙈